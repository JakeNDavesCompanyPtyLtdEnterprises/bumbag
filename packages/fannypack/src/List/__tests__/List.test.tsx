import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { List } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<List>Hello world</List>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <div>
        <List listStyleType="disc" listStylePosition="inside">
          <List.Item>Item 1</List.Item>
          <List.Item>Item 2</List.Item>
        </List>
        <List listStyleType="circle" listStylePosition="inside">
          <List.Item>Item 1</List.Item>
          <List.Item listStyleType="georgian">Item 2</List.Item>
          <List listStyleType="circle">
            <List.Item>Item 1</List.Item>
            <List.Item>Item 2</List.Item>
          </List>
        </List>
      </div>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a basic list', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly as an ordered list', () => {
    const { container } = render(
      <List isOrdered>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly as a horizontal list', () => {
    const { container } = render(
      <List isHorizontal>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<List use="div">Hello world</List>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with List props', () => {
      const { result } = renderHook(() => List.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<List>{ListProps => <div {...ListProps}>Hello world</div>}</List>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('List.root should render correctly', () => {
    const { container } = render(
      <List overrides={{ Link: { css: { root: { backgroundColor: 'red' } } } }}>hello world</List>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('List.root should render correctly', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { css: { root: { backgroundColor: 'red' } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.ordered should render correctly', () => {
    const { container } = render(
      <List isOrdered>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { css: { ordered: { backgroundColor: 'red' } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.ordered should render correctly', () => {
    const { container } = render(
      <List isHorizontal>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { css: { horizontal: { backgroundColor: 'red' } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('List.Item.root should render correctly', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { Item: { css: { root: { backgroundColor: 'red' } } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for isOrdered', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { defaultProps: { className: 'test', isOrdered: true } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for List.Item', () => {
    const { container } = render(
      <List>
        <List.Item>Item One</List.Item>
        <List.Item>Item Two</List.Item>
        <List.Item>Item Three</List.Item>
        <List.Item>Item Four</List.Item>
      </List>,
      {
        theme: { List: { Item: { defaultProps: { className: 'test' } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
