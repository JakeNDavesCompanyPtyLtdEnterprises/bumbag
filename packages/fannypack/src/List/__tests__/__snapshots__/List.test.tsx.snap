// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic list 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c1 {
  list-style: unset;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 .c0 {
  margin-left: 1rem;
}

<ul
  class="c0 c1 c2"
>
  <li
    class="c3 c2"
  >
    Item 1
  </li>
  <li
    class="c3 c2"
  >
    Item 2
  </li>
  <li
    class="c3 c2"
  >
    Item 3
  </li>
  <li
    class="c3 c2"
  >
    Item 4
  </li>
</ul>
`;

exports[`renders correctly for a list with styles 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c1 {
  list-style: unset;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 .c0 {
  margin-left: 1rem;
}

<div>
  <ul
    class="c0 c1 c2"
    style="list-style-type: disc; list-style-position: inside;"
  >
    <li
      class="c3 c2"
    >
      Item 1
    </li>
    <li
      class="c3 c2"
    >
      Item 2
    </li>
  </ul>
  <ul
    class="c0 c1 c2"
    style="list-style-type: circle; list-style-position: inside;"
  >
    <li
      class="c3 c2"
    >
      Item 1
    </li>
    <li
      class="c3 c2"
      style="list-style-type: georgian;"
    >
      Item 2
    </li>
    <ul
      class="c0 c1 c2"
      style="list-style-type: circle;"
    >
      <li
        class="c3 c2"
      >
        Item 1
      </li>
      <li
        class="c3 c2"
      >
        Item 2
      </li>
    </ul>
  </ul>
</div>
`;

exports[`renders correctly for horizontal lists 1`] = `
.c2 {
  margin: unset;
  pc1ng: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c4 .c0 {
  margin-left: 1rem;
}

.c5 .c0 {
  margin-left: 1rem;
}

.c5 .c0 {
  list-style-type: lower-alpha;
}

.c5 .c0 .c0 {
  list-style-type: lower-roman;
}

.c1 {
  list-style: unset;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 .c0 {
  margin-left: 1rem;
}

.c1 li {
  display: inline-block;
  margin-right: 1rem;
  margin-bottom: unset;
}

<div>
  <ul
    class="c0 c1 c2"
  >
    <li
      class="c3 c2"
    >
      Item 1
    </li>
    <li
      class="c3 c2"
    >
      Item 2
    </li>
    <li
      class="c3 c2"
    >
      Item 3
    </li>
    <li
      class="c3 c2"
    >
      Item 4
    </li>
  </ul>
</div>
`;

exports[`renders correctly for ordered lists 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin-bottom: 0.2rem;
}

.c4 .c0 {
  margin-left: 1rem;
}

.c1 {
  list-style: unset;
  list-style-type: none;
}

.c1:not(:last-child) {
  margin-bottom: 1rem;
}

.c1 .c0 {
  margin-left: 1rem;
}

.c1 {
  list-style-type: decimal;
}

.c1 .c0 {
  list-style-type: lower-alpha;
}

.c1 .c0 .c0 {
  list-style-type: lower-roman;
}

<div>
  <ol
    class="c0 c1 c2"
    style="list-style-position: inside;"
  >
    <li
      class="c3 c2"
    >
      Item 1
    </li>
    <li
      class="c3 c2"
    >
      Item 2
    </li>
    <ol
      class="c0 c1 c2"
    >
      <li
        class="c3 c2"
      >
        Item 3
      </li>
      <ol
        class="c0 c1 c2"
      >
        <li
          class="c3 c2"
        >
          Item 4
        </li>
      </ol>
    </ol>
  </ol>
</div>
`;
